### PR TITLE
US6008   settings backend

### PR DESCRIPTION
**Pull request by yujlin2**
Integration

### CODE REVIEW

File: src/components/AssignmentSettings.tsx [hunk 1]
```diff
@@ -195,7 +195,7 @@ function AssignmentSetting() {
 							.filter((course) =>
 								course.enrollments.some(
 									(enrollment) =>
-										enrollment.role === "StudentEnrollment"
+										enrollment.role === "TeacherEnrollment"
```
Comment (Stefce02) [inline src/components/AssignmentSettings.tsx:198]: Add comment to say that we're assuming subject coord will have TeachEnrollment. Can be changed to suit name of role in actual Canvas instance. Could make it a var in the env file

**Review by Stefce02 — CHANGES_REQUESTED** [review]
Aside from the comments their is one main issue:
- None of the requests show if the staff member is not allowed to add requests for even just  one assignment

I would also like to see:
- the setting be a radio button rather than checkbox: "allow all tutors to accept requests for this assignment?" - YES/NO. This should then remain ticked based on the user selection. If YES, the next time they go to it, it should still show YES. This button should also not show until the user has selected the assignment.
- Another change is to hide this setting tab all together for any staff member that is not TeacherEnrollment as it doesn't make sense for them to be able to see it as they are not the coordinator

File: src/components/StaffViewRequests.tsx [hunk 2]
```diff
@@ -67,36 +70,56 @@ function StaffViewRequests() {
...
+										hasPermission = course.id.toString() == assignmentSettingsData.courseId && assignmentSettingsData.tutorAccept && assignment?.id.toString() == assignmentSettingsData.assignmentId;
+									});
+	
+									return (
+										course.id.toString() ==
+											requestData.courseId &&
+										course.enrollments.some(
+											(enrollment) =>
+												enrollment.role ==
+													"TeacherEnrollment" ||
```
[the remark this answers is not in the document]
Comment (yujlin2) [reply src/components/StaffViewRequests.tsx:105]: I changed this because the TaEnrollment also satisfy this condition which would be incorrect.

**Review by Dragan14 — APPROVED** [review]
👍

**Pull request merged**